tutorials/pretrain_openwebtext.md

```diff
@@ -0,0 +1,139 @@
+
+You can easily change the size of the model by passing a different string to the model name variable
+
+```python
+model_name = "Llama-2-7b-hf"
+```
+
+at the top of this script.
+
+The currently supported model names are contained in the [config.py](https://github.com/Lightning-AI/lit-gpt/lit_gpt/config.py) file. 
+You can either search this file for lines containing "name =" or obtain the list of all supported models programmatically, as follows:
+
+```python
+from lit_gpt.config import configs
+
+for conf in configs:
+    print(conf["name"])
+```
+
+Keep in mind that the original LLaMA training for the 7B model required 83k A100 80GB
+hours (on a bigger dataset). However, for full pretraining on OpenWebText, you'll likely still need access to a cluster.
+
+Once you're in a cluster, you can follow [these instructions](https://lightning.ai/docs/fabric/stable/guide/multi_node/other.html)
+to launch the script across machines:
+
+- [SLURM cluster](https://lightning.ai/docs/fabric/stable/guide/multi_node/slurm.html)
+- [Barebones cluster](https://lightning.ai/docs/fabric/stable/guide/multi_node/barebones.html)
+- [MPI](https://lightning.ai/docs/fabric/stable/guide/multi_node/other.html)
+
+The script contains several configurations and hyperparameters you can tweak:
+
+```python
+out_dir = "out/training"
+save_interval = 1000
+eval_interval = 1000
+eval_iters = 100
+log_interval = 1
+
+# Hyperparameters
+learning_rate = 6e-4
+batch_size = 125
+micro_batch_size = 6
+max_iters = 600000  # num_epochs * (epoch_size // micro_batch_size) // devices
+weight_decay = 1e-1
+beta1 = 0.9
+beta2 = 0.95
+grad_clip = 1.0
+decay_lr = True
+warmup_iters = 2000
+lr_decay_iters = max_iters
+min_lr = 6e-5
+```
+
+For instance, `micro_batch_size` should be adjusted so the process will use the available
+GPU memory. For more tips to avoid out-of-memory issues, please also see the more detailed
+[Dealing with out-of-memory (OOM) errors](oom.md) guide.
+
+Last, logging is kept minimal in the script. In order to use a particular logger
+please refer to <https://lightning.ai/docs/fabric/stable/api/loggers.html> or
+call a logging client library like `wandb` directly.
+
+## Training a smaller model on a single GPU
+
+To train a smaller Pythia 70M model on a single GPU, you can modify the `pretrain/openwebtext.py` file to use the following settings:
+
+
+```python
+model_name = "Pythia-70m"
+```
+
+(Please see the the `download_*` scripts in the [../tutorials](../tutorials) for more information on downloading model checkpoints for different models.)
+
+Also, before you start training, note that you will need to prepare the dataset specifically for this model since it may use a different tokenizer:
+
+```bash
+python scripts/prepare_openwebtext.py \
+  --checkpoint_dir checkpoints/EleutherAI/pythia-70m/ \
+  --destination_path data/lit-openwebtext
+
+python pretrain/openwebtext.py \
+  --devices 4 \
+  --train_data_dir data/lit-openwebtext
+```
+
+## Using the Trainer Class
```

Comment:
```suggestion
## Using the Trainer
```
It's more like the trainer object that people use

Reply:
Or maybe saying "PyTorch Lightning `Trainer`" would be even better. There are thousands of Trainers out there, so better to be specific.